enhancements/insights/config-api.md

```diff
@@ -0,0 +1,296 @@
+
+## Summary
+
+This enhancement extends the `config.openshift.io` API with a new Insights API. The new API will enable a more appropriate way of configuring 
+as well as status reporting of the Insights operator. 
+The Insights operator is currently configured (for some historical reasons) via `support` secret (in the `openshift-config` namespace). 
+This secret is not present in a cluster by default and more importantly it is not the appropriate way to configure a cluster operator. 
+Providing a new configuration API for the Insights should bring a better user experience as well as easier way to introduce new configuration options (an example could be a way how to run only particular gatherers, which should help when running in non-standard environments as for example HyperShift) and the operator status reporting. 
+
+## Motivation
+
+The current configuration of the Insights operator is available via the `support` secret, which has to be created first and the configuration options include at least 5 items (upload endpoint, gathering interval, config related to SCA pull down) at the moment and there will be more in the future. 
+Configuring all this stuff via secret is not a convenient way to configure a cluster operator. There is also a configuration option on how to exclude/include some of the gatherers, but this option is currently hardcoded in the image (i.e not exposed to the user).  Exposing these existing options and defining new ones will increase overall user experience. 
+This would also allow to expose the basic Insights analysis results in the corresponding status section. This functionality now relies on the in-cluster Prometheus (and it will probably not be fully available in the HyperShift architecture).
+Finally, one of the reasons for this enhancement is to define a new way for a user to trigger the Insights data collection.  
+
+### Goals
+
+The goal of this enhancement is to propose a basic structure for a new Insights configuration API and define new configuration options. This includes following:
+
+- Move the majority of the current configuration options to a new configuration API
+- Define a new configuration option providing a way to trigger on demand Insights gathering
+- Define a new configuration option providing a way to include/exclude particular Insights data gatherers
+- Provide Insights status & reporting in the status field in the new API
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+The proposal is to introduce a new Insights `CustomResourceDefinition` in the OpenShift API providing a way to configure the Insights operator and to report its status. The new CRD should follow a common CRD structure providing `spec` and `status` attributes. 
+
+```go
+type InsightsSpec struct {
+	// Gathering spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	Gathering Gathering `json:"gathering,omitempty"`
```

Comment:
i'm not API expert, but we usually use something like `GatheringConfig` or `GatherConfig`

Reply:
OK why not. Thx